sdk/storage/src/common/crypt.cpp

```diff
@@ -82,96 +97,164 @@ namespace Azure { namespace Storage {
+      {
+        throw std::runtime_error("BCryptHashData failed");
+      }
 
-    status = BCryptHashData(
-        hashHandle,
-        reinterpret_cast<PBYTE>(const_cast<char*>(&text[0])),
-        static_cast<ULONG>(text.length()),
-        0);
-    if (!BCRYPT_SUCCESS(status))
-    {
-      throw std::runtime_error("BCryptHashData failed");
+      std::string hash;
+      hash.resize(AlgorithmProvider.HashLength);
+      status = BCryptFinishHash(
+          hashHandle, reinterpret_cast<PUCHAR>(&hash[0]), static_cast<ULONG>(hash.length()), 0);
+      if (!BCRYPT_SUCCESS(status))
+      {
+        throw std::runtime_error("BCryptFinishHash failed");
+      }
+
+      BCryptDestroyHash(hashHandle);
+
+      return hash;
     }
 
-    std::string hash;
-    hash.resize(AlgorithmProvider.HashLength);
-    status = BCryptFinishHash(
-        hashHandle, reinterpret_cast<PUCHAR>(&hash[0]), static_cast<ULONG>(hash.length()), 0);
-    if (!BCRYPT_SUCCESS(status))
+    std::string Hmac_Sha256(const std::string& text, const std::string& key)
     {
-      throw std::runtime_error("BCryptFinishHash failed");
-    }
 
-    BCryptDestroyHash(hashHandle);
+      static AlgorithmProviderInstance AlgorithmProvider(AlgorithmType::Hmac_Sha256);
 
-    return hash;
-  }
+      std::string context;
+      context.resize(AlgorithmProvider.ContextSize);
+
+      BCRYPT_HASH_HANDLE hashHandle;
+      NTSTATUS status = BCryptCreateHash(
+          AlgorithmProvider.Handle,
+          &hashHandle,
+          reinterpret_cast<PUCHAR>(&context[0]),
+          static_cast<ULONG>(context.size()),
+          reinterpret_cast<PUCHAR>(const_cast<char*>(&key[0])),
+          static_cast<ULONG>(key.length()),
+          0);
+      if (!BCRYPT_SUCCESS(status))
+      {
+        throw std::runtime_error("BCryptCreateHash failed");
+      }
+
+      status = BCryptHashData(
+          hashHandle,
+          reinterpret_cast<PBYTE>(const_cast<char*>(&text[0])),
+          static_cast<ULONG>(text.length()),
+          0);
+      if (!BCRYPT_SUCCESS(status))
+      {
+        throw std::runtime_error("BCryptHashData failed");
+      }
 
-  std::string Hmac_Sha256(const std::string& text, const std::string& key)
+      std::string hash;
+      hash.resize(AlgorithmProvider.HashLength);
+      status = BCryptFinishHash(
+          hashHandle, reinterpret_cast<PUCHAR>(&hash[0]), static_cast<ULONG>(hash.length()), 0);
+      if (!BCRYPT_SUCCESS(status))
+      {
+        throw std::runtime_error("BCryptFinishHash failed");
+      }
+
+      BCryptDestroyHash(hashHandle);
+
+      return hash;
+    }
+  } // namespace Details
+
+  struct Md5HashContext
   {
+    std::string buffer;
+    BCRYPT_HASH_HANDLE hashHandle = nullptr;
+    std::size_t hashLength = 0;
+  };
 
-    static AlgorithmProviderInstance AlgorithmProvider(AlgorithmType::Hmac_Sha256);
+  Md5::Md5()
+  {
+    static Details::AlgorithmProviderInstance AlgorithmProvider(Details::AlgorithmType::Md5);
```

Comment:
Why pascal case?

Reply:
Ditto for other local variables. I see snake case/pascal case/camel cases in different areas.

Reply:
fixed